### PR TITLE
AVS API 2023-03-01

### PR DESCRIPTION
**Pull request by cataggar**
These changes received [ARMSignedOff on March 29th](https://github.com/Azure/azure-rest-api-specs-pr/pull/9349#event-8871072727). Suppression was approved already [here](https://github.com/Azure/azure-rest-api-specs-pr/pull/9349#event-7636684643).

The primary changes are:
- `extendedNetworkBlocks` added to `PrivateCloudUpdateProperties`.
- The error response model is now using the `common-types/resource-management/v5/types.json#/definitions/ErrorResponse`.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[x] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version). Note that this doesn't apply if you are trying to merge a PR that was previously in the private repository.

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/overview_breakingchanges).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](http://aka.ms/aprwiki))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/vmware/resource-manager/Microsoft.AVS/stable/2023-03-01/skus.json
Comment (Hatim-ali95) [inline specification/vmware/resource-manager/Microsoft.AVS/stable/2023-03-01/skus.json:328]: @cataggar This would be a breaking change as we are returning sku.capacity as an integer in master (refer [this](https://msazure.visualstudio.com/DefaultCollection/One/_git/Azure-Dedicated-AVS?path=/src/conveyor/conveyor/AzureFSharp/Domain.fs&version=GBmaster&line=496&lineEnd=496&lineStartColumn=7&lineEndColumn=29&lineStyle=plain&_a=contents)). In this case it conflicts with this change as it defines capacity as a dictionary rather than an integer. 

I believe we Should be keeping the ARM Sku contract of "Microsoft.AVS/skus" API separate from the Sku definition required in the PrivateCloud / Cluster PUT/PATCH requests?

It has been done in the similar fashion for other RPs like Microsoft.Compute, Microsoft.apimanagement, etc where they have separate Sku definitions for {providerNamespace}/skus and the other resource specific APIs.
cc: @amtyagi

Comment (cataggar) [reply specification/vmware/resource-manager/Microsoft.AVS/stable/2023-03-01/skus.json:328]: In the current `2022-05-01` API version, `Sku` is defined as:
https://github.com/Azure/azure-rest-api-specs/blob/237afa17929313c909f72f2fea59f4437e305c99/specification/vmware/resource-manager/Microsoft.AVS/stable/2022-05-01/vmware.json#L6521-L6533
There is no definition of `capacity`.
The `Sku` definition here is the [ARM Sku Contract](https://armwiki.azurewebsites.net/api_contracts/SKUContract.html?tabs=resource-type-level). It is the same definition as before, but with more optional fields.

Comment (cataggar) [reply specification/vmware/resource-manager/Microsoft.AVS/stable/2023-03-01/skus.json:328]: The link you provided is an internal implementation detail. Internally, it may make sense to use different types. We could restore `Sku` and rename the new type to `ResourceSku` [like Microsoft.Compute](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/compute/resource-manager/Microsoft.Compute/Skus/stable/2021-07-01/skus.json), but I'm not sure the benefit. Are you wanting to return `capacity` with an operation like `PrivateCloud_Get`? We don't currently return it.

Comment (Hatim-ali95) [reply specification/vmware/resource-manager/Microsoft.AVS/stable/2023-03-01/skus.json:328]: @cataggar, not with the PrivateCloud but we are going to return the capacity field as part of the Cluster_Get response with the Autoscale feature and also would be honouring the same in the Cluster PUT/PATCH calls. It would be better if we rename the new type to ResourceSku and extend the existing Sku type for the optional integer capacity field. Let me know if that makes sense and whether you are aligned with the thought. We can extend the existing Sku type in a separate swagger PR though.

Comment (Hatim-ali95) [reply specification/vmware/resource-manager/Microsoft.AVS/stable/2023-03-01/skus.json:328]: The  support for capacity field for Clusters is added in the September API version:

https://msazure.visualstudio.com/DefaultCollection/One/_git/Azure-Dedicated-AVS?path=/src/conveyor/conveyor/ConveyorHost/WebApi/Microsoft.AVS/ResourceDefinitions/PrivateClouds_Clusters/2023-09-01/Serialization.fs&version=GBmaster&line=180&lineEnd=180&lineStartColumn=22&lineEndColumn=70&lineStyle=plain&_a=contents

Comment (cataggar) [reply specification/vmware/resource-manager/Microsoft.AVS/stable/2023-03-01/skus.json]: I removed the sku changes d4a6b866c420b2805ff6f79307e55e19f8f7930c and we can flush out the details for the next API version.

Comment (Hatim-ali95) [reply specification/vmware/resource-manager/Microsoft.AVS/stable/2023-03-01/skus.json]: Sounds good. Will you be rebasing the September branch (https://github.com/Azure/azure-rest-api-specs-pr/tree/AVS/2023-09-01) with these changes so that we can introduce the Sku changes required for Autoscale?

File: specification/vmware/resource-manager/Microsoft.AVS/stable/2023-03-01/vmware.json
Comment (Hatim-ali95) [inline specification/vmware/resource-manager/Microsoft.AVS/stable/2023-03-01/vmware.json:316]: Looks like this is missed as part of the undo.

**Comment by raosuhas** [comment]
@cataggar is there any diff between the checked in version in the private repo and this that needs to be reviewed ?

**Comment by raosuhas** [comment]
> @cataggar is there any diff between the checked in version in the private repo and this that needs to be reviewed ?

Can you please respond to this ?

**Comment by cataggar** [comment]
> @cataggar is there any diff between the checked in version in the private repo and this that needs to be reviewed ?

@raosuhas No, there are not. All changes have been reviewed already in the private repo.

**Comment by visingla-ms** [comment]
> > @cataggar is there any diff between the checked in version in the private repo and this that needs to be reviewed ?
> 
> @raosuhas No, there are not. All changes have been reviewed already in the private repo.

Approving based this Private Repo PR: https://github.com/Azure/azure-rest-api-specs-pr/pull/9349/.

**Comment by cataggar** [comment]
@rkmanda, anything left before you merge this?

**Comment by kazrael2119** [comment]
this js sdk breaking may be caused by this pr with change ErrorResponse from v1 to v5 in common-type

**Comment by cataggar** [comment]
> this js sdk breaking may be caused by this pr with change ErrorResponse from v1 to v5 in common-type

That breaking API change was approved.  @kazrael2119, do you have more details specific to the JS SDK?

**Comment by cataggar** [comment]
I [posted in Teams](https://teams.microsoft.com/l/message/19:0351f5f9404446e4b4fd4eaf2c27448d@thread.skype/1690982782259?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1690982782259&teamName=Azure%20SDK&channelName=API%20Spec%20Review&createdTime=1690982782259&allowXTenantAccess=false) looking to get assistance for merging this.

**Comment by rkmanda** [comment]
This still needs a suppression review which I will do shortly

**Comment by rkmanda** [comment]
For the suppression applied, the ask is not to change the existing pattern but to document any pattern that the service is already honoring. Yes this documentation will trigger a breaking change but thats easily explainable to get an approval.

**Comment by cataggar** [comment]
@rkmanda, I've added the resource name patterns as requested. CI looks happy.

**Comment by cataggar** [comment]
Can this be merged now or do you need to start a breaking change review for specifying the resource name patterns?

**Comment by cataggar** [comment]
I need approval for the Go SDK breaking changes at https://github.com/azure-sdk/azure-sdk-for-go/pull/6280 . Can someone please approve?

**Review by rkmanda — APPROVED** [review]

**Comment by cataggar** [comment]
The Go SDK has approved the breaking change. I think that was the last holdup. Please merge.

**Pull request merged**
